library/ssl_tls13_client.c

```diff
@@ -1341,14 +1334,8 @@ static int ssl_write_key_shares_ext( mbedtls_ssl_context *ssl,
 
         p += len;
         *olen += len + 2;
-        MBEDTLS_SSL_DEBUG_ECP( 3, "ECDHE: Q ", &ssl->handshake->ecdh_ctx[nr].Q );
+        MBEDTLS_SSL_DEBUG_ECP( 3, "ECDHE: Q ", &ssl->handshake->ecdh_ctx.Q );
 
-        nr++;
```

Comment:
For this loop,  `nr` is removed , should we reconsider about the loop ? I guess there are some problem now.

Reply:
Yes, we need to remove the loop here. The previous code would offer as many key shares as there are supported groups. This isn't suitable for constrained devices, and we instead fall back to (a) still offering the same number of groups, but (b) only offering one key share a time.

By default, it makes sense to use the 0-th group for the initial key share, because that's supposed to be the most preferred one.

After receiving an HRR, however, we might need to change which share we offer.

I would therefore suggest that you introduce an index variable like `offered_group` which defaults to `0` but can be reconfigured during the handshake, and which determines which entry from the `key_share_curves_list` we're using for the key share.

Does that make sense?

Reply:
Thanks. I will update this part later